contracts/Witnet.sol

```diff
@@ -70,13 +101,76 @@ library Witnet {
+    if (error[0] <= uint8(ErrorCodes.Size)) {
+      errorCode = ErrorCodes(error[0]);
+    } else {
+      errorCode = ErrorCodes.Unknown;
+    }
+
+    if (errorCode == ErrorCodes.SourceScriptNotCBOR) {
+      errorMessage = abi.encodePacked("Source script #", utoa(error[1]), " was not a valid CBOR value");
+    } else if (errorCode == ErrorCodes.SourceScriptNotArray) {
+      errorMessage = abi.encodePacked("The CBOR value in script #", utoa(error[1]), " was not an Array of calls");
+    } else if (errorCode == ErrorCodes.SourceScriptNotRADON) {
+      errorMessage = abi.encodePacked("The CBOR value in script #", utoa(error[1]), " was not a valid RADON script");
+    } else if (errorCode == ErrorCodes.RequestTooManySources) {
+      errorMessage = abi.encodePacked("The request contained too many sources (", utoa(error[1]), ")");
+    } else if (errorCode == ErrorCodes.ScriptTooManyCalls) {
+      errorMessage = abi.encodePacked("Script #", utoa(error[2]), " from the ", stageName(error[1]), " stage contained too many calls (", utoa(error[3]), ")");
+    } else if (errorCode == ErrorCodes.UnsupportedOperator) {
+      errorMessage = abi.encodePacked("Operator code ", utoa(error[4]), " found at call #", utoa(error[3]), " in script #", utoa(error[2]), " from ", stageName(error[1]), " stage is not supported");
+    } else if (errorCode == ErrorCodes.HTTP) {
+      errorMessage = abi.encodePacked("Source #", utoa(error[1]), " could not be retrieved. Failed with HTTP error code: ", utoa(error[2] / 100), utoa(error[2] % 100 / 10), utoa(error[2] % 10));
+    } else if (errorCode == ErrorCodes.Underflow) {
+      errorMessage = abi.encodePacked("Underflow at operator code ", utoa(error[4]), " found at call #", utoa(error[3]), " in script #", utoa(error[2]), " from ", stageName(error[1]), " stage");
+    } else if (errorCode == ErrorCodes.Overflow) {
+      errorMessage = abi.encodePacked("Overflow at operator code ", utoa(error[4]), " found at call #", utoa(error[3]), " in script #", utoa(error[2]), " from ", stageName(error[1]), " stage");
+    } else if (errorCode == ErrorCodes.DivisionByZero) {
+      errorMessage = abi.encodePacked("Division by zero at operator code ", utoa(error[4]), " found at call #", utoa(error[3]), " in script #", utoa(error[2]), " from ", stageName(error[1]), " stage");
+    } else if (errorCode == ErrorCodes.RuntimeError) {
+      errorMessage = abi.encodePacked("Unspecified runtime error at operator code ", utoa(error[4]), " found at call #", utoa(error[3]), " in script #", utoa(error[2]), " from ", stageName(error[1]), " stage");
+    } else if (errorCode == ErrorCodes.InsufficientConsensusError) {
+      errorMessage = abi.encodePacked("Insufficient consensus. Required: ", utoa(error[1]), ". Achieved: ", utoa(error[2]));
+    } else {
+      errorMessage = abi.encodePacked("Unknown error (", utoa(error[0]), ")");
+    }
```

Comment:
Knowing that the formatting rule is a real pain...
`in case of more than 3 arguments, drop each into its own line.`